src/slider/slider.component.ts

```diff
@@ -0,0 +1,328 @@
+		const input = this.elementRef.nativeElement.querySelector("input:not([type=range])");
+		if (input) {
+			input.disabled = v;
+		}
+	}
+
+	get disabled() {
+		return this._disabled;
+	}
+	/** Emits every time a new value is selected */
+	@Output() valueChange: EventEmitter<number> = new EventEmitter();
+	@HostBinding("class.bx--slider-container") hostClass = true;
+	@ViewChild("thumb") thumb: ElementRef;
+	@ViewChild("track") track: ElementRef;
+	@ViewChild("range") range: ElementRef;
+
+	public bottomRangeId = `${this.id}-bottom-range`;
+	public topRangeId = `${this.id}-top-range`;
+
+	protected isMouseDown = false;
+	/** Array of event subscriptions so we can batch unsubscribe in `ngOnDestroy` */
+	protected eventSubscriptions: Array<Subscription> = [];
+	protected slidAmount = 0;
+	protected input: HTMLInputElement;
+	protected _value = 0;
+	protected _disabled = false;
+
+	constructor(protected elementRef: ElementRef) {}
+
+	ngAfterViewInit() {
+		// bind mousemove and mouseup to the document so we don't have issues tracking the mouse
+		this.eventSubscriptions.push(fromEvent(document, "mousemove").subscribe(this.onMouseMove.bind(this)));
+		this.eventSubscriptions.push(fromEvent(document, "mouseup").subscribe(this.onMouseUp.bind(this)));
+
+		// ODO: ontouchstart/ontouchmove/ontouchend
+
+		// set up the optional input
+		this.input = this.elementRef.nativeElement.querySelector("input:not([type=range])");
+		if (this.input) {
+			this.input.type = "number";
+			this.input.classList.add("bx--slider-text-input");
+			this.input.classList.add("bx--text-input");
+			this.input.setAttribute("aria-labelledby", `${this.bottomRangeId} ${this.topRangeId}`);
+			this.input.value = this.value.toString();
+			// bind events on our optional input
+			this.eventSubscriptions.push(fromEvent(this.input, "change").subscribe(this.onChange.bind(this)));
+			this.eventSubscriptions.push(fromEvent(this.input, "focus").subscribe(this.onFocus.bind(this)));
+		}
+	}
+
+	/** Clean up our DOMEvent subscriptions */
+	ngOnDestroy() {
+		this.eventSubscriptions.forEach(subscription => {
+			subscription.unsubscribe();
+		});
+	}
+
+	/** Send changes back to the model */
+	propagateChange = (_: any) => { };
+
+	/** Register a change propagation function for `ControlValueAccessor` */
+	registerOnChange(fn: any) {
+		this.propagateChange = fn;
+	}
+
+	/** Callback to notify the model when our input has been touched */
+	onTouched: () => any = () => { };
+
+	/** Register a callback to notify when our input has been touched */
+	registerOnTouched(fn: any) {
+		this.onTouched = fn;
+	}
+
+	/** Receives a value from the model */
+	writeValue(v: any) {
+		this.value = v;
+	}
+
+	/** Returns the amount of "completeness" as a fraction of the total track width */
+	getPercentComplete() {
+		const trackWidth = this.track.nativeElement.getBoundingClientRect().width;
+		return this.slidAmount / trackWidth;
```

Comment:
If I'm reading this right, this might have to be multiplied by a 100, or function should change the name.
```suggestion
		return this.slidAmount / trackWidth * 100;
```

Reply:
Percent, fraction, it's all the same right? 😉 

(fixed)